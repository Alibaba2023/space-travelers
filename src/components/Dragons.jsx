import React from 'react';
import '../style/dragons.css';

const Dragons = () => (
  <div className="container">Dragons</div>
);

export default Dragons;
