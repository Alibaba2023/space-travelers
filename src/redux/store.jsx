import { applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { configureStore } from '@reduxjs/toolkit';
import dragonsSlice from './dragons/dragonsSlice';
import missionSlice from './missions/missionSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    dragon: dragonsSlice,
    missions: missionSlice,
    rockets: rocketsSlice,
  },
},
applyMiddleware(logger));

export default store;
