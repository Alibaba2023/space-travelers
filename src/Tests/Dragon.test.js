import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Dragons from '../components/Dragons';

test('renders loading state correctly', () => {
  render(
    <Provider store={store}>
      <Dragons />
    </Provider>,
  );

  const h2Element = screen.getByText(/Loading/i);
  const loadingElement = screen.getByTestId('h2test');

  expect(loadingElement).toBeInTheDocument();
  expect(h2Element).toBeInTheDocument();
});
