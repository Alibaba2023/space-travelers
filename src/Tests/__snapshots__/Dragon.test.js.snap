// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component renders correctly in DragonTest 1`] = `
<div
  className="spinner-border"
  role="status"
/>
`;
